netweaver/extract_nw_archives.sls

```diff
@@ -0,0 +1,51 @@
+{%- from "netweaver/map.jinja" import netweaver with context -%}
+{% set nw_extract_dir = netweaver.nw_extract_dir %}
+{% set swpm_extract_dir = nw_extract_dir~'/SWPM' %}
+
+setup_nw_extract_directory:
+  file.directory:
+    - name: {{ nw_extract_dir }}
+    - mode: 755
+    - makedirs: True
+
+{% if netweaver.sapcar_exe_file is defined and netweaver.swpm_sar_file is defined %}
+
+extract_installer_file:
+    sapcar.extracted:
+    - name: {{ netweaver.swpm_sar_file }}
+    - sapcar_exe: {{ netweaver.sapcar_exe_file }}
+    - output_dir: {{ nw_extract_dir }}/SWPM
+    - options: "-manifest SIGNATURE.SMF"
+
+{% endif %}
+
+{% set additional_dvd_folders = [] %}
+
+{% for dvd in netweaver.additional_dvds %}
+{% set dvd_folder = dvd.split('.')[0] %}
```

Comment:
This split looks dangerous, what if the file is `x.tar.gz` or similar. We should check if jinja offers some way to get the path without the extension

Reply:
Yes I checked for alternatives but could not find something in jinja.
We are extracting mainly the rar,exe,zip this is fine for that

Reply:
Other alternative I found is to use the salt's jinja filter for regex https://docs.saltstack.com/en/latest/topics/jinja/index.html#regex-match